### PR TITLE
MES-2813: Poll test criteria into users DDB table

### PR DESCRIPTION
**Pull request by johnmorleybjss**
# Description and relevant Jira numbers
* Extend MySQL examiner query to also join the `DES_TEST_CRITERIA` table
* Make an additional query to obtain "universal" permissions applying to examiner
* Break out a dedicated row mapper to handle converting examiner query results to `StaffDetail`
* Add a new field to `StaffDetail` recording the `testPermissionPeriods`
* Add missing `DES_TEST_CRITERIA` table creation to TARS replica schema script

```json
{
  "role": "DE",
  "staffNumber": "01234567",
  "testPermissionPeriods": [
    {
      "from": "2019-07-03",
      "testCategory": "B",
      "to": "2019-08-01"
    },
    {
      "from": "2019-08-30",
      "testCategory": "B",
      "to": null
    }
  ]
}
```

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [x] Reviewers selected in Github
- [x] PR link added to JIRA

### CODE REVIEW

File: destination-db/database-objects/010_createTables.sql [hunk 1]
```diff
@@ -294,4 +294,11 @@ CREATE TABLE IF NOT EXISTS TEST_CATEGORY
 	TEST_CATEGORY_REF VARCHAR(10) NOT NULL,
 	THEORY_TYPE_CODE INT NOT NULL,
 	CONSTRAINT PK_TEST_CATEGORY PRIMARY KEY (TEST_CATEGORY_REF)
+);
+
+CREATE TABLE IF NOT EXISTS DES_TEST_CRITERIA (
```
Comment (chrisnappin) [inline destination-db/database-objects/010_createTables.sql]: EXAMINER_STAFF_NUMBER and WITH_EFFECT_TO are nullable

Comment (johnmorleybjss) [reply destination-db/database-objects/010_createTables.sql]: Marked the inverse as `NOT NULL`.

**Comment by johnmorleybjss** [comment]
Good spot @chrisnappin, this doesn't handle universal permissions yet. I'll look into how we can extend the functionality.

**Comment by johnmorleybjss** [comment]
The universal permissions functionality is now implemented @chrisnappin.

**Pull request merged**